[[Bug 18305]] Add Synonyms to Dictionary

bwmilby opened the pull request:
LC 7 allowed searching for synonyms within the dictionary.  Implementation appears to have been by duplicating the dictionary entries for each synonym.  This fix will duplicate the dictionary entry for each synonym when the entries are built from the repository.  "Synonym of" is added to the display name of each non-glossary entry (glossary synonyms do not seem to mean that the terms mean the same thing - i.e. card button, card field, card control, card object).

The change is made in the "revDocsParseDictionaryToLibraryArray" handler.  On each pass through the loop, the resulting array is tested to see if the "Synonyms" key contains data.  If it does, then loop through each synonym and duplicate the entry (taking into account that glossary entries specify themselves as a synonym).  The display name is changed to the synonym and "Synonym of ..." is added as a cross-reference for non-glossary entries.

bwmilby committed:
[[Bug 18305]] Add Synonyms to Dictionary

LC 7 allowed searching for synonyms within the dictionary.  Implementation appears to have been by duplicating the dictionary entries for each synonym.  This fix will duplicate the dictionary entry for each synonym when the entries are built from the repository.  "Synonym of" is added to the display name of each non-glossary entry (glossary synonyms do not seem to mean that the terms mean the same thing - i.e. card button, card field, card control, card object).

The change is made in the "revDocsParseDictionaryToLibraryArray" handler.  On each pass through the loop, the resulting array is tested to see if the "Synonyms" key contains data.  If it does, then loop through each synonym and duplicate the entry (taking into account that glossary entries specify themselves as a synonym).  The display name is changed to the synonym and "Synonym of ..." is added as a cross-reference for non-glossary entries.

livecodeali commented:
Hi @bwmilby, thanks very much for the fix and very thorough commit message! It looks good to me. Regarding your second point, yes it perhaps does need something like what you mention there - although I'm slightly wary of something called `display syntax` being essentially made unfit for display (I know it wouldn't currently matter as only the first element of the array is displayed, but this might change at some point).

I think the best option would be to make the dictionary search use display name as well as display syntax in its search - along with this commit that would solve the problem.

bwmilby committed:
[[Bug 18305]] Add synonym to 'display syntax'

Update the bug fix to add the synonym to the "display syntax" element.  Each dictionary entry only carries the first "syntax" entry over to the "display syntax" as [1].  Using [2] for the synonym entry (it does not get displayed in the dictionary window).

bwmilby commented:
Just saw your comment after I committed a change with the 'display syntax' addition.  Not sure how to back that commit out though.

mwieder commented:
@bwmilby I wouldn't back out the commit, just change it locally and push again.
If you do need to back it out, then git reset locally and force push the change.

bwmilby commented:
Pull #1658 created on livecode-ide to add the display name to the search in the IDE.

bwmilby committed:
Undo the "display syntax" change

Back out the "display syntax" addition to the dictionary.  Pull request #1658 created on the IDE to update the dictionary search to include the "display name".

livecodeali commented:
@livecode-vulcan review ok 941f789

livecodeali commented:
@bwmilby We can squash these commits when merging so that it comes out as a single commit anyway. Thanks very much!

livecode-vulcan commented:
💙 review by @livecodeali ok 941f7890a3d4beffcb8b1824ed2d5aa7a66b02fe

livecode-vulcan commented:
😎 test success 941f7890a3d4beffcb8b1824ed2d5aa7a66b02fe

* try-community-armv6-android-api8: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/457)
* try-community-armv6-android-api9: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/465)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/444)
* try-community-universal-ios-iphoneos10.3: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos10.3/builds/387)
* try-community-universal-ios-iphonesimulator10.3: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator10.3/builds/387)
* try-community-universal-mac-macosx10.6: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/452)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/457)
* try-community-x86-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/455)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/468)
* try-community-x86_64-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/455)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/464)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/482)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/451)
